src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -36,15 +40,16 @@ const SearchPage = () => {
   `;
 
   const [searchText, setSearchText] = useState('');
-  const [results, setResults] = useState([]);
+  const [results, setResults] = useState(undefined);
+  const [filter, setFilter] = useState('author');
 
-  const [executeSearch, { data }] = useLazyQuery(SEARCH_QUERY, {
-    // setting author variable here to use in the query above
+  const [executeSearch, { loading, data }] = useLazyQuery(SEARCH_QUERY, {
+    // Setting author variable here to use in the query above
     variables: { author: searchText },
-
     onCompleted: () => {
       // We only want to display a feed once, this would avoid the issue if
       // there are multiple posts by the same author returned in our query
+      setResults(undefined);
       const finalResults = data.getPosts
         .filter(
```

Comment:
I don't understand this filter.  Can you comment above it?

Reply:
I guess the most relevant example would be right now where Krystyna has two posts back to back, if we were to display the results without doing this filter, we'd have two <AuthorResult /> component with Krystyna as opposed to one. We only want unique feed results here.